File: Ilxomjonov_Olimjon/core/proxy/security_proxy.dart
import '../../modules/security/security.dart';

class SecurityProxy {
  final Security _security = Security();
  bool isAdmin = false;

  void activateCamera() {
    if (isAdmin) {
      _security.activateCameras();
    } else {
      print('UR NOT AN ADMIN');
    }
  }

  void deactivateCameras() {
    if (isAdmin) {
      _security.deactivateCameras();
    } else {
      print('UR NOT AN ADMIN');
    }
  }
}

File: saydullojonrakhmonov/src/core/proxy/SecurityProxy.ts
import type { ISecuritySystem } from '../../modules/security/index.ts';

export class SecurityProxy implements ISecuritySystem {
  private realSystem: ISecuritySystem;
  private isAuthorized: boolean = false;

  constructor(realSystem: ISecuritySystem) {
    this.realSystem = realSystem;
  }

  login() {
    this.isAuthorized = true;
    console.log("\n Admin logged in \n");
  }

  activateCameras() {
    if (this.isAuthorized) {
      this.realSystem.activateCameras();
    } else {
      console.log("\n Access denied: Login required \n");
    }
  }

  detectThreat() {
    this.realSystem.detectThreat(); 
  }
}